From 28be9bed70efd3055a3380d5c9547a0ae5a59f88 Mon Sep 17 00:00:00 2001
From: Marc Mao <marc.mao@intel.com>
Date: Tue, 20 Nov 2012 09:59:42 +0800
Subject: [PORT FROM MAIN] Change the framebuffer format to BGRA base on PVR default setting

BZ: 69058

In MR1, the framebuffer format is hard coded as ARGB8888. But in
IMG's default setting, it can't support HAL RGBA as egl render
surface format. Although we can fix it in PVR code, we will not do
that because it need change many place and that is not a validate
path.

Change-Id: Id5bf55ab5615e4ee7df6b444ae906b0937631e40
Orig-Change-Id: I4b17e3a254bd6b7c82c412cfb5fc4cf9baf8df62
Signed-off-by: Marc Mao <marc.mao@intel.com>
Reviewed-on: http://android.intel.com:8080/74735
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Xu, Randy <randy.xu@intel.com>
Tested-by: Xu, Randy <randy.xu@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 .../surfaceflinger/DisplayHardware/HWComposer.cpp  |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/services/surfaceflinger/DisplayHardware/HWComposer.cpp b/services/surfaceflinger/DisplayHardware/HWComposer.cpp
index 3f00f07..d8a0eb1 100644
--- a/services/surfaceflinger/DisplayHardware/HWComposer.cpp
+++ b/services/surfaceflinger/DisplayHardware/HWComposer.cpp
@@ -364,7 +364,7 @@ status_t HWComposer::queryDisplayProperties(int disp) {
     }
 
     // FIXME: what should we set the format to?
-    mDisplayData[disp].format = HAL_PIXEL_FORMAT_RGBA_8888;
+    mDisplayData[disp].format = HAL_PIXEL_FORMAT_BGRA_8888;
     mDisplayData[disp].connected = true;
     if (mDisplayData[disp].xdpi == 0.0f || mDisplayData[disp].ydpi == 0.0f) {
         float dpi = getDefaultDensity(h);
@@ -730,7 +730,7 @@ int HWComposer::getVisualID() const {
         // FIXME: temporary hack until HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED
         // is supported by the implementation. we can only be in this case
         // if we have HWC 1.1
-        return HAL_PIXEL_FORMAT_RGBA_8888;
+        return HAL_PIXEL_FORMAT_BGRA_8888;
         //return HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED;
     } else {
         return mFbDev->format;
-- 
1.7.4.1

